eclipse-collections/src/main/java/org/eclipse/collections/impl/multimap/AbstractSynchronizedMultimap.java

```diff
@@ -131,7 +131,7 @@ public boolean putAllPairs(Pair<K, V>... pairs)
     }
 
     @Override
-    public boolean putAllPairs(Iterable<Pair<K, V>> pairs)
+    public boolean putAllPairs(Iterable<? extends Pair<? extends K, ? extends V>> pairs)
```

Comment:
I have mixed feelings about `? extends Pair`. I guess it's technically correct.

It would be good to see the unit tests change to use some supertype/subtype interaction. Maybe something like a Map with Number keys where we add a Pair<Integer, V>

Reply:
even though it is technically correct, do you think it is functionally correct to have `? extends Pair` ?

Reply:
I think it is functionally correct as Collections generally allow to add subclass objects.
Please see use case in AbstractMutableMultimapTestCase.java.